Remove environment from payment_method

cbrunsdon opened the pull request:
This anti-feature adds complexity to both the code and the payment
method interface. End-users don't (and shouldn't) understand what this
does it does not follow any standard rails conventions.

Removing this "feature" is a net benefit to the codebase.

Commit:
Remove environment from payment_method

This anti-feature adds complexity to both the code and the payment
method interface. End-users don't (and shouldn't) understand what this
does it does not follow any standard rails conventions.

Removing this "feature" is a net benefit to the codebase.

Commit:
Remove guide references to payment_method env

Functionality was removed in the previous commit

TeatroIO commented:
I've prepared a stage. <a href='http://freerunningtech-kill-payment-method-environment.spree-spree-9ed724b77adec04c49c2.ttrcloud.com' target='_blank'>Click to open.</a>

JDutil commented:
Should have a migration removing the column.

BenMorganIO commented:
I think we should do this for `Spree::Tracker` as well.

BenMorganIO commented:
It was from this anti-feature that I had to stop work on #5770.

JDutil commented:
That's a good point as well.  With the introduction of a Store model now in core it really makes more sense to have a "staging" or "production" store than environment allowing you to achieve this sort of feature across everything really.

gmacdougall commented:
As part of the migration, it should set `active: false` for all PaymentMethod's which do not match the current environment. This will be to avoid allowing the bogus environment to work in production for someone who previously had the bogus payment enabled for development mode in their production database.

This should also have an entry in the Changelog.

gmacdougall commented:
@JDutil I disagree that the Store should contain a production/staging environment. I feel that the environment is specified by the database that is being used. The production database contains production orders, and configuration. The staging database contains staging orders and configuration.

The place to deal with this is on the point where the data is inserted, either through a migration, initializer, or through the admin interface.

JDutil commented:
I'm not saying that a store should contain an environment setting, but pointing out that they could be used as such.

JDutil commented on core/app/models/spree/payment_method.rb:
```diff
@@ -29,13 +27,12 @@ def payment_source_class
     def self.available(display_on = 'both')
       all.select do |p|
```
I think we could could accomplish this with arel instead of a select.

cbrunsdon replied:
I am not really aiming to change any of the logic or structure with this PR, but I have a plan to tackle the display_on madness next.

SpreeCI replied:
thats fine wanted to point out I dislike this method anyway

JDutil replied:
Well I dislike the method, and following links not in default browser...

BenMorganIO replied:
What is @SpreeCI?

JDutil replied:
Just a user account I didn't mean to comment with. It's used to signup for services such as circleci under a spree account rather than my own.

Commit:
Remove the environment column from payment_method

Sets any payment method not in the current environment to be inactive.
Can't be reversed as we have set the active = false

cbrunsdon commented:
Thanks for the feedback folks, updated the PR to include a migration removing the column and updating all Payment method's from different environments to be false.
